pkg/plugin/plugin_test.go

```diff
@@ -20,211 +21,52 @@ func TestCreateOrUpdate(t *testing.T) {
-		},
-		{
-			name:     "waitforinfra: deployments error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepWaitForInfraDeployments,
-		},
-		{
-			name:     "ConsoleHealth: error",
-			isUpdate: true,
-			wantErr:  true,
-			errStep:  api.PluginStepWaitForConsoleHealth,
+	}
+	deployer := func(ctx context.Context, azuretemplate map[string]interface{}) error {
+		return nil
+	}
+	cs := &api.OpenShiftManagedCluster{
+		Properties: api.Properties{
+			AgentPoolProfiles: []api.AgentPoolProfile{
+				{Role: api.AgentPoolProfileRoleMaster, Name: "master"},
+				{Role: api.AgentPoolProfileRoleCompute, Name: "compute"},
+				{Role: api.AgentPoolProfileRoleInfra, Name: "infra"},
+			},
 		},
 	}
+
 	for _, tt := range tests {
 		mockGen := mock_arm.NewMockGenerator(mockCtrl)
-		mockGen.EXPECT().Generate(nil, nil, "", tt.isUpdate, gomock.Any()).Return(nil, nil)
-		if tt.wantErr {
-			err := &api.PluginError{Err: fmt.Errorf("test error"), Step: tt.errStep}
-			switch tt.errStep {
-			case api.PluginStepDeploy, api.PluginStepInitialize,
-				api.PluginStepWaitForConsoleHealth, api.PluginStepWaitForNodes,
-				api.PluginStepUpdateMasterAgentPoolListVMs,
-				api.PluginStepUpdateMasterAgentPoolReadBlob, api.PluginStepUpdateMasterAgentPoolDrain,
-				api.PluginStepUpdateMasterAgentPoolDeallocate, api.PluginStepUpdateMasterAgentPoolUpdateVMs,
-				api.PluginStepUpdateMasterAgentPoolReimage, api.PluginStepUpdateMasterAgentPoolStart,
-				api.PluginStepUpdateMasterAgentPoolWaitForReady, api.PluginStepUpdateMasterAgentPoolUpdateBlob,
-				api.PluginStepUpdateWorkerAgentPoolListVMs, api.PluginStepUpdateWorkerAgentPoolReadBlob,
-				api.PluginStepUpdateWorkerAgentPoolWaitForReady, api.PluginStepUpdateWorkerAgentPoolUpdateBlob:
-				if tt.isUpdate {
-					mockUp.EXPECT().Update(nil, nil, nil, nil, gomock.Any()).Return(err)
-				} else {
-					mockUp.EXPECT().Deploy(nil, nil, nil, nil, gomock.Any()).Return(err)
-				}
-			case api.PluginStepWaitForWaitForOpenShiftAPI:
-				if tt.isUpdate {
-					mockUp.EXPECT().Update(nil, nil, nil, nil, gomock.Any()).Return(nil)
-				} else {
-					mockUp.EXPECT().Deploy(nil, nil, nil, nil, gomock.Any()).Return(nil)
-				}
-				mockUp.EXPECT().WaitForInfraServices(nil, nil).Return(nil)
-				mockUp.EXPECT().HealthCheck(nil, nil).Return(err)
-			case api.PluginStepWaitForInfraDaemonSets, api.PluginStepWaitForInfraDeployments:
-				if tt.isUpdate {
-					mockUp.EXPECT().Update(nil, nil, nil, nil, gomock.Any()).Return(nil)
-				} else {
-					mockUp.EXPECT().Deploy(nil, nil, nil, nil, gomock.Any()).Return(nil)
-				}
-				mockUp.EXPECT().WaitForInfraServices(nil, nil).Return(err)
-			}
+		mockGen.EXPECT().Generate(nil, cs, "", tt.isUpdate, gomock.Any()).Return(nil, nil)
+		mockUp.EXPECT().CreateClients(nil, cs).Return(nil)
+		mockUp.EXPECT().Initialize(nil, cs).Return(nil)
+		if tt.isUpdate {
+			mockUp.EXPECT().UpdateMasterAgentPool(nil, cs, &cs.Properties.AgentPoolProfiles[0]).Return(nil)
```

Comment:
I'm interested to know whether it's plausible to strengthen this as in:
```golang
c := foo.EXPECT().Bar().Return(baz)
c = bar.EXPECT().Baz().Return(bash).After(c)
c = baz.EXPECT().Bash().Return(qux).After(c)
```
(again as in update_master_test.go)
or whether we think it'll make us go mad.  Thoughts?

Reply:
done, easy enough